pkg/visibility/api/rest/pending_workload_CQ.go

```diff
@@ -0,0 +1,99 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1alpha1 "sigs.k8s.io/kueue/apis/visibility/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/constants"
+	"sigs.k8s.io/kueue/pkg/queue"
+
+	_ "k8s.io/metrics/pkg/apis/metrics/install"
+)
+
+type pendingWorkloadsInCqREST struct {
+	kueueMgr *queue.Manager
+	log      logr.Logger
+}
+
+var _ rest.Storage = &pendingWorkloadsInCqREST{}
+var _ rest.GetterWithOptions = &pendingWorkloadsInCqREST{}
+var _ rest.Scoper = &pendingWorkloadsInCqREST{}
+var _ rest.SingularNameProvider = &pendingWorkloadsInCqREST{}
+
+func NewPendingWorkloadsInCqREST(kueueMgr *queue.Manager) *pendingWorkloadsInCqREST {
+	return &pendingWorkloadsInCqREST{
+		kueueMgr: kueueMgr,
+		log:      ctrl.Log.WithName("pending-workload-in-cq"),
+	}
+}
+
+// New implements rest.Storage interface
+func (m *pendingWorkloadsInCqREST) New() runtime.Object {
+	return &v1alpha1.PendingWorkloadList{}
+}
+
+// Destroy implements rest.Storage interface
+func (m *pendingWorkloadsInCqREST) Destroy() {}
+
+// Get implements rest.Getter interface
+// It fetches information about pending workloads and returns according to query params
+func (m *pendingWorkloadsInCqREST) Get(ctx context.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	pendingWorkloadOpts, ok := opts.(*v1alpha1.PendingWorkloadOptions)
+	if !ok {
+		return nil, fmt.Errorf("invalid options object: %#v", opts)
+	}
+	limit := pendingWorkloadOpts.Limit
+	offset := pendingWorkloadOpts.Offset
+
+	var wls []v1alpha1.PendingWorkload
+	for index, val := range m.kueueMgr.GetPendingWorkloadsInfo(name) {
+		if index >= int(offset) && index < int(offset)+int(limit) {
+			wls = append(wls, v1alpha1.PendingWorkload{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      val.Name,
+					Namespace: val.Namespace,
+				},
+			})
+		}
+	}
+	return &v1alpha1.PendingWorkloadList{Items: wls}, nil
+}
+
+// NewGetOptions creates a new options object
+func (m *pendingWorkloadsInCqREST) NewGetOptions() (runtime.Object, bool, string) {
+	// If no query parameters were passed the generated defaults function are not executed so it's necessary to set default values here as well
+	return &v1alpha1.PendingWorkloadOptions{
+		Limit: constants.DefaultPendingWorkloadsLimit,
+	}, false, ""
+}
+
+// NamespaceScoped implements rest.Scoper interface
+func (m *pendingWorkloadsInCqREST) NamespaceScoped() bool {
+	return false
+}
+
+// GetSingularName implements rest.SingularNameProvider interface
+func (m *pendingWorkloadsInCqREST) GetSingularName() string {
+	return "pendingWorkloadsInCQ"
```

Comment:
just to make sure we are being consistent... which name does the pods/log subresource use?

Reply:
All subresources use a name inherited from the parent resource. This name is used to create correct reports and to access resources from kubectl. Hence, I'll move it to the `CqREST` implementation. Even if it may not be directly used there, it will span across all its subresources which may be useful.